components/motor/gpio/motor_encoder_test.go

```diff
@@ -4,755 +4,279 @@ import (
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 0)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encA.Tick(context.Background(), true, nowNanosTest()), test.ShouldBeNil) // 10 -> 11
+	t.Run("encoded motor test GoFor forward", func(t *testing.T) {
+		test.That(t, m.goForInternal(context.Background(), 10, 1, 1), test.ShouldBeNil)
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 2)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 1)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encB.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil) // 11 -> 01
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 3)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 1)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encA.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil) // 01 -> 00
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 4)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 2)
+			on, powerPct, err := m.IsPowered(context.Background(), nil)
+			test.That(tb, on, test.ShouldBeTrue)
+			test.That(tb, powerPct, test.ShouldBeGreaterThan, 0)
 			test.That(tb, err, test.ShouldBeNil)
 		})
 	})
 
-	t.Run("motor encoder move backward", func(t *testing.T) {
-		th := setup(t)
-		defer th.Teardown()
-
-		enc := th.Encoder
-		encA := th.EncoderA
-		encB := th.EncoderB
-
-		// this should do nothing because it's the initial state
-		test.That(t, encA.Tick(context.Background(), false, nowNanosTest()), test.ShouldBeNil)
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 0)
-		})
-		test.That(t, encB.Tick(context.Background(), true, nowNanosTest()), test.ShouldBeNil) // we go from state 00 -> 10
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			pos := enc.RawPosition()
-			test.That(tb, pos, test.ShouldEqual, 1)
-			posFl, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
-			pos = int64(posFl)
-			test.That(tb, pos, test.ShouldEqual, 0)
-			test.That(tb, err, test.ShouldBeNil)
-		})
-
-		test.That(t, encA.Tick(context.Background(), true, nowNanosTest()), test.ShouldBeNil) // 10 -> 11
+	t.Run("encoded motor test GoFor backwards", func(t *testing.T) {
+		test.That(t, m.goForInternal(context.Background(), -10, -1, -1), test.ShouldBeNil)
```

Comment:
[note to self] this test confuses me abit, since I would expect a negative rpm & goal to result in a positive direction. 

after looking more, goForInternal receives the result from goForMath, so this function makes sense after learning what goForMath outputs